synapse/handlers/directory.py

```diff
@@ -252,12 +253,14 @@ async def get_association(self, room_alias: RoomAlias) -> JsonDict:
                     retry_on_dns_fail=False,
                     ignore_backoff=True,
                 )
+            except RequestSendFailed:
+                raise SynapseError(502, "Failed to fetch alias")
             except CodeMessageException as e:
                 logging.warning("Error retrieving alias")
                 if e.code == 404:
                     fed_result = None
                 else:
-                    raise
+                    raise SynapseError(502, "Failed to fetch alias")
```

Comment:
Shouldn't we log the original error as well? (or maybe it's already logged as part of the HTTP client logs, I can't remember)

Reply:
Good Q. We should log response success/errors in the http clients, so I don't think its necessary here.

Reply:
FWIW, re-raising here implicitly adds `from e`, so the traceback of the original exception will be available to anyone who catches it.